cmd/service-controller/controller.go

```diff
@@ -188,7 +188,14 @@ func NewController(cli *client.VanClient, origin string, tlsConfig *certs.TlsCon
 	}
 
 	if enableSkupperEvents {
-		controller.eventHandler = kube.NewSkupperEventRecorder(cli.Namespace, cli.KubeClient)
+		deployment, _ := kube.GetDeployment(types.ControllerDeploymentName, cli.Namespace, cli.KubeClient)
```

Comment:
If this results in an error, deployment might be nil and the subsequent line would cause a crash. Might be nicer to fail with a clear error?

Reply:
This is being executed inside the service-controller deployment that is running, so it should not be nil...Nevertheless,  the client could fail for any other reason, so I agree on that.

Reply:
updated in 42c9c91129661fda0d7cd12114be47febf6e5fa7